Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,213 @@ namespace xr
                     float rawPose[7]{};
                     ArHitResult_getHitPose(xrContext->Session, hitResult, tempPose);
                     ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
-                    HitResult hitResult{};
-                    RawToPose(rawPose, hitResult.Pose);
+                    HitResult hitTestResult{};
+                    RawToPose(rawPose, hitTestResult.Pose);
 
-                    hitResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
-                    filteredResults.push_back(hitResult);
+                    hitTestResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
+                    filteredResults.push_back(hitTestResult);
                     frameTrackables.push_back(trackable);
                 }
             }
         }
 
+        // Converts an image bitmap to grayscale assumes an RGB8 or RGB8A image for use with ARCore.
+        void ConvertBitmapToGrayscale(
+            const uint8_t* imagePixelBuffer,
+            const int32_t width,
+            const int32_t height,
+            const int32_t stride,
+            uint8_t* grayscaleBuffer)
+        {
+            const uint32_t pixelStride = stride / width;
+            for (int h = 0; h < height; ++h)
+            {
+                for (int w = 0; w < width; ++w)
+                {
+                    const uint8_t* pixel{&imagePixelBuffer[w * pixelStride + h * stride]};
+                    const uint8_t r{*pixel};
+                    const uint8_t g{*(pixel + 1)};
+                    const uint8_t b{*(pixel + 2)};
+                    grayscaleBuffer[w + h * width] = static_cast<uint8_t>(0.213f * r + 0.715 * g + 0.072 * b);
+                }
+            }
+        }
+
+        std::vector<std::string> CreateAugmentedImageDatabase(const std::vector<System::Session::Frame::ImageTrackingRequest>& requests)
```

Comment:
It seems like this is the only place the input vector is marked as const. Should it be const everywhere?